barretenberg/cpp/src/barretenberg/vm2/tracegen/lib/lookup_into_unary.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <cassert>
+
+#include "barretenberg/vm2/tracegen/lib/lookup_builder.hpp"
+
+namespace bb::avm2::tracegen {
+
+template <typename LookupSettings> class LookupIntoUnary : public BaseLookupTraceBuilder<LookupSettings> {
+  private:
+    // This is an efficient implementation of indexing into the precomputed table.
+    uint32_t find_in_dst(const std::array<FF, LookupSettings::LOOKUP_TUPLE_SIZE>& tup) const override
+    {
+        // clk is the index
+        const auto& [clk, _] = tup;
```

Comment:
I see we are repeating ourselves here, it shouldn't be difficult to create one for CLK indexing that can be used for all. Will do later.